kotlin/workflow-ui-android/src/test/java/com/squareup/workflow/ui/WorkflowRunnerViewModelTest.kt

```diff
@@ -44,13 +59,40 @@ class WorkflowRunnerViewModelTest {
     scope.coroutineContext[Job]!!.invokeOnCompletion { e ->
       if (e is CancellationException) cancelled = true
     }
-    val runner = WorkflowRunnerViewModel(viewRegistry, emptyFlow(), emptyFlow(), scope)
 
-    assertThat(cancelled).isFalse()
-    runner.output.test()
+    val runner = WorkflowRunnerViewModel(viewRegistry, emptyFlow(), MutableLiveData(), scope)
+
     assertThat(cancelled).isFalse()
 
     runner.clearForTest()
     assertThat(cancelled).isTrue()
   }
+
+  @Test fun magilla() {
+    val outputs = ConflatedBroadcastChannel<String>()
+
+    val w = object : StatelessWorkflow<Unit, String, Unit>() {
+      override fun render(
+        input: Unit,
+        context: RenderContext<Nothing, String>
+      ) {
+        context.onWorkerOutput(outputs.asWorker()) { emitOutput(it) }
+      }
+    }
+
+    val factory = WorkflowRunnerViewModel.Factory(
+        w, viewRegistry, flowOf(Unit), null, TestCoroutineDispatcher()
+    )
+    val model = factory.create(WorkflowRunnerViewModel::class.java)
+
+    runBlockingTest {
+
+      model.output.observeForever {
+        assertThat(it).isEqualTo("Snot")
+      }
+
+      // java.lang.RuntimeException: Method getMainLooper in android.os.Looper not mocked. See http://g.co/androidstudio/not-mocked for details.
```

Comment:
This is what I hit in #450. The exception says we can just set the main dispatcher override statically, but I would want to understand where the hard-coding is coming from first.

Reply:
Exactly.

Reply:
I suspect this is going to make us abandon `androidx.lifecycle.CoroutineLiveDataKt#liveData`. Should figure out what it's doing and file a bug.